text/0000-RouteInfo-Metadata.md

```diff
@@ -0,0 +1,210 @@
+- Start Date: 2018-11-02
+- RFC PR: (leave this empty)
+- Ember Issue: (leave this empty)
+
+# RouteInfo MetaData
+
+## Summary
+
+The RFC introduces the ability to associate application specific metadata with it's corresponding `RouteInfo` object. This also adds a `metadata` field to the `RouteInfo` which will be the return value of `buildRouteInfoMetaData` for it's corresponding `Route`.
+
+```js
+// app/route/profile.js
+import Route from '@ember/routing/route';
+
+export default Route.extend({
+
+  buildRouteInfoMetaData() {
+    return {
+      trackingKey: 'page_profile',
+      profile: {
+        viewing: this.userId,
+        locale: this.userLocale
+      }
+    }
+  }
+  // ...
+});
+```
+
+```js
+// app/services/analytics.js
+import Service, { inject } from '@ember/service';
+
+export default Service.extend({
+  router: inject('router'),
+  init() {
+    this._super(...arguments);
+    this.router.on('routeDidUpdate', (transition) => {
+      let { to, from } = transition;
+      let { metadata: fromMeta } = from;
```

Comment:
haha, I hate destructuring like this 😸 

I think this is simpler, shorter, and easier:

```js
let fromMeta = from.metadata;
```